### DISCUSSION TITLE
v0.1.0 vs v0.2.0

### DISCUSSION THREAD

**Discussion by galanisd** [eclipse-edc/Connector#3392] [General]
We have a java project that is based on EDC. 
When we are using  EDC 0.1.0 (as dependency) which actually uses v2 asset API everything works as expected.
We are sending a JSON_LD and by using breakpoints we see that it is received at the createAsset endpoint exactly as sent

![image](https://github.com/eclipse-edc/Connector/assets/10739759/f9db773d-9501-440f-84ae-268fdd396d89)

When we switch to  0.2.0 (or 0.2.1) and send the same JSON_LD the createAsset endpoint (v2 or v3) receives something different 

![image](https://github.com/eclipse-edc/Connector/assets/10739759/856f13f9-130b-4229-a25c-41394f24cca1)

For example @context has been removed and EDC has ceated arrays.

In all cases we inspect the received JSON-LD by adding a breakpoint at 
public JsonObject createAsset(...)

and we check the content of JsonObject.

Any idea what is happening? It is very weird.

**Comment by ndr-brt** [comment]
Could you report the call you are doing and posting the json response? I'm not understanding a lot from the attached screenshots.
BTW, `@context` has not been removed, it's a fundamental node in a json-ld document, and every json produced by the management api correctly specifies the `@context`

**Comment by galanisd** [comment]
We are posting something like the following. 
I ommited some parts to make it shorter

```
{
  "@context": {
    "edc": "https://w3id.org/edc/v0.0.1/ns/",
    "dcat": "http://www.w3.org/ns/dcat/",
    "dct": "https://purl.org/dc/terms/",
    "adms": "https://www.w3.org/TR/vocab-adms/",
    "ms": "http://w3id.org/meta-share/meta-share/",
    "loc": "http://id.loc.gov/vocabulary/iso639-1/",
    "iana": "http://www.iana.org/assignments/media-types/application/",
    "rdf": "http://www.w3.org/1999/02/22-rdf-syntax-ns#",
    "xsd": "http://www.w3.org/2001/XMLSchema#",
    "foaf": "http://xmlns.com/foaf/0.1/",
    "@base": "https://lds.eu/",
    "omtd": "http://w3id.org/meta-share/omtd-share/"
  },
  "@id": "asset-id2",
  "edc:properties": {
    "@type": "edc:Asset",
    "dcat:version": "1.0.0",
    "dct:title": [
      {
        "@value": "title",
        "@language": "en"
      }
    ],
    "ms:domain": {
      "@id": "http://publications.europa.eu/resource/authority/data-theme/AGRI"
    },
    "dct:alternative": {
      "@value": "alternative",
      "@language": "en"
    },
    "dct:description": [
      {
        "@value": "a nice description for a very nice corpus",
        "@language": "en"
      },
      {
        "@value": "ένα πολύ ωραίο σύνολο δεδομένων",
        "@language": "el"
      }
    ],
    "dct:identifier": "testId1",
    "adms:identifier": "testId3",
    "dct:publisher": {
      "@id": "ILSP",
      "@type": "foaf:Agent",
      "foaf:name": [
        {
          "@value": "ILSP",
          "@language": "en"
        },
        {
          "@value": "ILSPs",
          "@language": "el"
        }
      ],
      "dct:identifier": "testId1"
    },
    ....
  },
  "edc:dataAddress": {
    "@type": "edc:DataAddress",
    "edc:name": "Asset Name",
    "edc:baseUrl": {
      "@value": "https://jsonplaceholder.typicode.com/users",
      "@type": "xsd:anyURI"
    },
    "edc:type": "HttpData"
  }
}
```

It seems that 0.1.0 gets exactly what we sent at the createAsset endpoint whilst in 0.2.1 the message 
has been changed before createAsset; i.e., the namepace prefixes have been somehow already been replaced with the respective URI based on @context and @context has been removed. See the screenshot below from my IDE (yellow popup window).

![image](https://github.com/eclipse-edc/Connector/assets/10739759/aba973be-4650-4c07-9da5-a5b80742284f)

If I copy paste the received JSON  to an JSON editor it looks like 

![image](https://github.com/eclipse-edc/Connector/assets/10739759/316adcb2-fa97-4d17-90e8-296195cbb1a1)


In any case last week we adapted our connector to the new logic of 0.2.1

**Comment by galanisd** [comment]
If the behavior that I describe is the expected ... I will close the discussion

**Reply by paullatzelsperger** [reply]
@galanisd you may not be aware that all our APIs perform JSON-LD expansion upon ingress. What you are seeing is the [expanded form](https://www.w3.org/TR/json-ld11/#expanded-document-form) and it is not only expected, it is by design.

**Comment by galanisd** [comment]
Ok. Then what we get is what is expected. Closing the discussion. Thanks.
